llvm/lib/Target/Sparc/SparcSubtarget.h

```diff
@@ -29,6 +29,12 @@ namespace llvm {
 class StringRef;
 
 class SparcSubtarget : public SparcGenSubtargetInfo {
+  // Reserve*Register[i] - *#i is not available as a general purpose register.
+  BitVector ReserveGRegister;
```

Comment:
Can this be a single `BitVector` indexed by physical register number? This would simplify `TRI.getReservedRegs()`, I think.

Reply:
I was thinking about something like
’"ReservedRegisters[SP::G0 + " # i # "]"’. Would that work?